fairlearn/metrics/_multi_curve_plot.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+from ..utils._input_validation import (
+    _validate_and_reformat_input,
+    _INCONSISTENT_ARRAY_LENGTH,
+    _INPUT_DATA_FORMAT_ERROR_MESSAGE,
+)
+from ..postprocessing._plotting import _MATPLOTLIB_IMPORT_ERROR_MESSAGE
+from ._make_derived_metric import _DerivedMetric
+from typing import Callable, Union
+from numpy import amax, amin, array
+from sklearn.utils.validation import check_array
+from numpy import zeros
+
+
+def plot_model_comparison(
+    *,
+    x_axis_metric: Callable[..., Union[float, int]],
+    y_axis_metric: Callable[..., Union[float, int]],
+    y_true,
+    y_preds,
+    sensitive_features,
+    ax=None,
+    axis_labels=True,
+    point_labels=None,
+    model_kwargs=None,
+    legend=False,
+    legend_kwargs={},
+    plot=True,
+    **kwargs,
+):
+    r"""
+    Plot a model comparison.
+
+    Parameters
+    ----------
+    x_axis_metric : Callable
+        The (aggregating) metric function for the x-axis
+        The passed metric function must take `y_true, y_pred`, and optionally `sensitive_features`.
+        If the metric is grouped, it must aggregate results. For instance, use
+        `make_derived_metric(metric=balanced_accuracy_score, transform='group_min')`
+        to aggregate the `balanced_accuracy_score`.
+
+    y_axis_metric : Callable
+        The (aggregating) metric function for the y-axis, similar to x_axis_metric.
+        The passed metric function must take `y_true, y_pred`, and optionally `sensitive_features`.
+        If the metric is grouped, it must aggregate results.
+
+    y_true : List, pandas.Series, numpy.ndarray, pandas.DataFrame
+        The ground-truth labels (for classification) or target values (for regression).
+
+    y_preds : array-like
+        An array-like containing predictions per model. Hence, predictions of
+        model :code:`i` should be in :code:`y_preds[i]`.
+
+    sensitive_features : List, pandas.Series, dict of 1d arrays, numpy.ndarray, pandas.DataFrame, optional
+        The sensitive features which should be used to create the subgroups.
+        At least one sensitive feature must be provided.
+        All names (whether on pandas objects or dictionary keys) must be strings.
+        We also forbid DataFrames with column names of ``None``.
+        For cases where no names are provided
+        we generate names ``sensitive_feature_[n]``.
+
+    ax : matplotlib.axes.Axes, optional
+        If supplied, the scatter plot is drawn on this Axes object.
+        Else, a new figure with Axes is created.
+
+    axis_labels : bool
+        If true, add the names of x and y axis metrics
+
+    point_labels : list
```

Comment:
I think this should be either `bool` or a list of strings; if False, nothing is shown, if True default labels are shown, and if the list is provided then the strings in the list are used.

In addition, there should be a similar `legend_labels` field that would accomplish the same for the legend (see a worked out example below in my comment on `model_kwargs`).

Reply:
Related to `point_labels`, I think we should add an argument something like `point_label_position` which would take a list of two floats as value and would work to create an offset for all the point labels. I know that this was discussed earlier (and the decision was perhaps not to do it), but without the ability to specify the position, the point label functionality is not something that I would want to use. Whenever I used point labels in example notebooks, I made sure to position them that they are legible (see, e.g., the example in my model_kwargs comment). The goal would be to apply this same offset to all points... this is consistent with how the `position` argument is used in [pd.DataFrame.plot](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.plot.html), see the example of that usage here:

![image](https://user-images.githubusercontent.com/20565444/168609823-4c96868c-6c86-461d-86a5-97f2ab546be8.png)

**VERSUS here**

![image](https://user-images.githubusercontent.com/20565444/168609546-9153ecb1-c4d0-4015-8b4e-e20c4ffbb42a.png)